Foundation/NSString.swift

```diff
@@ -405,15 +405,39 @@ extension NSString {
     }
     
     public func rangeOfCharacterFromSet(searchSet: NSCharacterSet) -> NSRange {
-        NSUnimplemented()
+        return rangeOfCharacterFromSet(searchSet, options: [])
     }
     
     public func rangeOfCharacterFromSet(searchSet: NSCharacterSet, options mask: NSStringCompareOptions) -> NSRange {
-        NSUnimplemented()
+        return rangeOfCharacterFromSet(searchSet, options: mask, range: NSMakeRange(0, length))
     }
     
     public func rangeOfCharacterFromSet(searchSet: NSCharacterSet, options mask: NSStringCompareOptions, range searchRange: NSRange) -> NSRange {
-        NSUnimplemented()
+        if mask.contains(.RegularExpressionSearch) {
+            NSUnimplemented()
+        }
+        if searchRange.length == 0 {
+            return NSMakeRange(NSNotFound, 0)
+        }
+        
+#if os(Linux)
+        var cfflags = CFStringCompareFlags(mask.rawValue)
+        if mask.contains(.LiteralSearch) {
+            cfflags |= UInt(kCFCompareNonliteral)
```

Comment:
probably these constants would be good to expose as a uniform re-export on OS X/iOS targets so that the linux (and potentially other platforms) would all have a similar interface.